linera-core/src/client/validator_manager/mod.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! This module manages communication with validator nodes, including connection pooling,
+//! load balancing, request deduplication, caching, and performance tracking.
+
+mod cache;
+mod in_flight_tracker;
+mod manager;
+mod node_info;
+mod request;
+mod scoring;
+
+pub use manager::ValidatorManager;
+pub use request::RequestKey;
+pub use scoring::ScoringWeights;
+
+// Module constants
+pub(super) const MAX_IN_FLIGHT_REQUESTS: usize = 100;
+pub(super) const MAX_ACCEPTED_LATENCY_MS: f64 = 5000.0;
+pub(super) const CACHE_TTL_SEC: u64 = 2;
+pub(super) const CACHE_MAX_SIZE: usize = 1000;
+// Maximum latency for an in-flight request before we stop deduplicating it.
+pub(super) const MAX_REQUEST_TTL_MS: u64 = 200;
```

Comment:
Should be CLI options and/or env variables, I guess.

Reply:
Done  in [Move all ValidatorManager config consts to ClientContextOptions](https://github.com/linera-io/linera-protocol/pull/4752/commits/59e2df164f7cf0c5d0cfaa3d4ae978feffff1de5)